### DISCUSSION TITLE
Can bpf_prog_test_run be facilitated without an extension driver

### DISCUSSION THREAD

**Discussion by shankarseal** [microsoft/ebpf-for-windows#3045] [Ideas]
Extension drivers have to implement create/destroy context functions for bpf_prog_test_run. Can this be done by extension developers publishing some sort of type description, that can be used by eBPF core to de-serialize the buffer sent by the user mode to construct the program context? This bpf_prog_test_run can be executed without any real extension installed at all.

**Comment by saxena-anurag** [comment]
The eBPF program may be invoking some program type specific helper functions. That may not work with the above proposal.
